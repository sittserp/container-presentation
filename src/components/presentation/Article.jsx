import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ author, title, description, url, urlToImage }) => (
  <>
    <figure>
      <img src={urlToImage} />
      <figcaption>{title, author}</figcaption>
    </figure>
    <p>{description}</p>
    <p>{url}</p>
  </>
);

Article.propTypes = {
  author: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  urlToImage: PropTypes.string.isRequired
};

export default Article;
