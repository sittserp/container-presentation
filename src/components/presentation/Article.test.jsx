import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Article from './Article';

describe('Article component', () => {
  afterEach(() => cleanup());
  it('renders Article', () => {
    const { asFragment } = render(<Article 
      author="dummy data author" 
      title="dummy data title"
      description="dummy data description"
      url="dummy data url"
      urlToImage="dummy data urlToImage"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
