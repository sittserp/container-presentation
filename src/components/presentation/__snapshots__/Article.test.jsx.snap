// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders Article 1`] = `
<DocumentFragment>
  <figure>
    <img
      src="dummy data urlToImage"
    />
    <figcaption>
      dummy data author
    </figcaption>
  </figure>
  <p>
    dummy data description
  </p>
  <p>
    dummy data url
  </p>
</DocumentFragment>
`;
